### ISSUE TITLE
Does not work on macOS?

### ISSUE THREAD

**Issue by messense** [kkysen/pyenv-python#1]
`python --version` just hangs without any output.

**Comment by messense** [comment]
```bash
➜  ~ env | grep PYENV
PYENV_SHELL=zsh
PYENV_VIRTUALENV_INIT=1
PYENV_VIRTUALENV_DISABLE_PROMPT=1
➜  ~ ls .pyenv
cache    shims    versions
```

**Comment by kkysen** [comment]
There was a bug with running the system `python` when `pyenv` doesn't have a local or global version set.  This `python` would default to finding the system `python` by searching in `$PATH` for the next `python` that's not itself, but with `pyenv` installed, this would be `$PYENV_ROOT/shims/python`, which would then try to find the system `python` itself, which would end up being this `python`, creating an infinite loop.

I fixed it (see the latest commit, <https://github.com/kkysen/pyenv-python/commit/52316312d60b5d8916a81fdd50397429595d9a37>) by skipping `$PYENV_ROOT/shims/python` in `$PATH` as well, so now it should work.  Let me know if the fix works.

Also, it shouldn't have anything to do with macOS.  I tried to write everything portably, but I don't have a mac to test it on so I can't be certain.  I've only tested it on WSL Ubuntu.
